pennylane/math/single_dispatch.py

```diff
@@ -51,7 +51,11 @@ def _i(name):
 ar.register_function("numpy", "coerce", lambda x: x)
 ar.register_function("numpy", "block_diag", lambda x: _scipy_block_diag(*x))
 ar.register_function("builtins", "block_diag", lambda x: _scipy_block_diag(*x))
-ar.register_function("numpy", "gather", lambda x, indices: x[np.array(indices)])
+ar.register_function(
+    "numpy",
+    "gather",
+    lambda x, indices, axis=0: x[:, np.array(indices)] if axis == 1 else x[np.array(indices)],
+)

```

Comment:
Remember that the axis could be any positive integer, not just zero and one. If we are trying to replicate the behavior of `tf.gather` across all interfaces, we also need to support cases like `gather(x, 0, axis=4)`

Reply:
I copied this code from [_qubit_device.py](https://github.com/PennyLaneAI/pennylane/blob/92c7ae784c59019cd03d63d5b9235bbceb6747b9/pennylane/_qubit_device.py#L116-L118), buuuut you're right that we can do better. I think the equivalent functions are actually `np.take_along_axis`, and jax and torch have their own gather methods. I don't need this anymore (nor does anyone) so I feel less motivated to test and add it to our repo. Perhaps this PR should just be closed : )